apps/aecore/src/aec_peers_pool.erl

```diff
@@ -0,0 +1,1695 @@
+%%  - `all` + `unverified`: Returns the total number of unverified peers.
+%%  - `available` + `both`: Returns the number of peers selectable peers.
+%%  - `available` + `verified`: Returns the number of selectable verified peers.
+%%  - `available` + `unverified`: Returns the number of selectable unverified peers.
+%%
+%% Note that the result of `available` peers do not take into account any
+%% additional restrictions that could be given to `select/3` or `select/4`.
+%% In addition, peers currently in standby that may have exhausted there standby
+%% time will **not** be counted.
+-spec count(state(), all | available, select_target()) -> non_neg_integer().
+count(St, all, both) ->
+  ?ST{verif_pool = VerifPool, unver_pool = UnverPool} = St,
+  pool_size(VerifPool) + pool_size(UnverPool);
+count(St, all, verified) ->
+  ?ST{verif_pool = VerifPool} = St,
+  pool_size(VerifPool);
+count(St, all, unverified) ->
+  ?ST{unver_pool = UnverPool} = St,
+  pool_size(UnverPool);
+count(St, available, both) ->
+  ?ST{lookup_verif = VerifLookup, lookup_unver = UnverLookup} = St,
+  lookup_size(VerifLookup) + lookup_size(UnverLookup);
+count(St, available, verified) ->
+  ?ST{lookup_verif = VerifLookup} = St,
+  lookup_size(VerifLookup);
+count(St, available, unverified) ->
+  ?ST{lookup_unver = UnverLookup} = St,
+  lookup_size(UnverLookup).
+
+%% @doc Returns where the given peer identifier is pooled and if it is available.
+-spec peer_state(state(), peer_id()) ->
+  {verified | unverified | undefined, boolean() | undefined}.
+peer_state(St, PeerId) ->
+  ?ST{standby = Standby} = St,
+  case find_peer(St, PeerId) of
+    undefined -> {undefined, undefined};
+    Peer ->
+      IsAvailable = not (peer_is_selected(Peer) or maps:is_key(PeerId, Standby)),
+      {peer_state(Peer), IsAvailable}
+  end.
+
+%% @doc Returns if the given peer identifier is pooled as verified.
+%% If given an unknown identifier it returns `undefined`.
+-spec is_verified(state(), peer_id()) -> undefined | boolean().
+is_verified(St, PeerId) ->
+  case find_peer(St, PeerId) of
+    undefined -> undefined;
+    Peer -> peer_state(Peer) =:= verified
+  end.
+
+%% @doc Returns if the given peer identifier is pooled as unverified.
+%% If given an unknown identifier it returns `undefined`.
+-spec is_unverified(state(), peer_id()) -> undefined | boolean().
+is_unverified(St, PeerId) ->
+  case find_peer(St, PeerId) of
+    undefined -> undefined;
+    Peer -> peer_state(Peer) =:= unverified
+  end.
+
+%% @doc Returns if the given peer identifier is available for selection.
+%% If given an unknown identifier it returns `undefined`.
+-spec is_available(state(), peer_id()) -> undefined | boolean().
+is_available(St, PeerId) ->
+  ?ST{standby = Standby} = St,
+  case find_peer(St, PeerId) of
+    undefined -> undefined;
+    Peer -> not (peer_is_selected(Peer) or maps:is_key(PeerId, Standby))
+  end.
+
+%% @doc Updates/inserts a peer.
```

Comment:
The documentation should also explain _why_ and _when_ the function should be called.

Reply:
Not sure what you mean. The doc state that this is used to add a peer to the pool or refresh it. What do you think is missing ?